consensus/src/main/java/org/apache/iotdb/consensus/multileader/MultiLeaderServerImpl.java

```diff
@@ -479,6 +481,45 @@ public void notifyPeersToRemoveSyncLogChannel(Peer targetPeer)
     }
   }
 
+  public void waitTargetPeerUntilSyncLogCompleted(Peer targetPeer)
+      throws ConsensusGroupAddPeerException {
+    long checkIntervalInMs = 10_000L;
+    try (SyncMultiLeaderServiceClient client =
+        syncClientManager.borrowClient(targetPeer.getEndpoint())) {
+      while (true) {
+        TWaitSyncLogCompleteRes res =
+            client.waitSyncLogComplete(
+                new TWaitSyncLogCompleteReq(targetPeer.getGroupId().convertToTConsensusGroupId()));
+        if (res.complete) {
+          logger.info(
+              "{} SyncLog is completed. TargetIndex: {}, CurrentSyncIndex: {}",
+              targetPeer,
+              res.searchIndex,
+              res.safeIndex);
+          return;
+        }
+        logger.info(
+            "{} SyncLog is still in progress. TargetIndex: {}, CurrentSyncIndex: {}",
+            targetPeer,
+            res.searchIndex,
+            res.safeIndex);
+        Thread.sleep(checkIntervalInMs);
+      }
+    } catch (IOException | TException e) {
+      throw new ConsensusGroupAddPeerException(
+          String.format(
+              "error when waiting %s to complete SyncLog. %s", targetPeer, e.getMessage()),
+          e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new ConsensusGroupAddPeerException(
+          String.format(
+              "thread interrupted when waiting %s to complete SyncLog. %s",
+              targetPeer, e.getMessage()),
+          e);
```

Comment:
I think we shouldn't throw `ConsensusGroupAddPeerException` here which may cause confuse.

Reply:
Sure, will define and use another exception